packages/eslint-plugin/src/stylex-no-unused.js

```diff
@@ -0,0 +1,287 @@
+        // stylex.create can only be singular variable declarations at the root
+        // of the file so we can look directly on Program and populate our set.
+        node.body
+          .filter(({ type }) => type === 'VariableDeclaration')
+          .map(({ declarations }) =>
+            declarations && declarations.length === 1 ? declarations[0] : null,
+          )
+          .filter(Boolean)
+          .filter(({ init }) => init && isStylexDeclaration(init))
+          .forEach(saveStylexCalls);
+      },
+
+      // Exempt used styles: "stylex.__" or "styles[__]"
+      MemberExpression(node: MemberExpression) {
+        if (
+          node.object.type === 'Identifier' &&
+          stylexProperties.has(node.object.name)
+        ) {
+          if (node.computed && node.property.type !== 'Literal') {
+            stylexProperties.delete(node.object.name);
+          } else if (node.property.type === 'Identifier') {
+            stylexProperties.get(node.object.name)?.delete(node.property.name);
+          } else if (node.property.type === 'Literal') {
+            stylexProperties.get(node.object.name)?.delete(node.property.value);
+          }
+        }
+      },
+      // catch function call "functionName(param)"
+      CallExpression(node: CallExpression) {
+        const functionName = node.callee?.name;
+        if (functionName == null || !stylexProperties.has(functionName)) {
+          return;
+        }
+        const namespaces = stylexProperties.get(functionName);
+        if (namespaces == null) {
+          return;
+        }
+        node.arguments?.forEach(checkArguments(namespaces));
+      },
+
+      // Exempt used styles: export const exportStyles = stylex.create({});
+      ExportNamedDeclaration(node: ExportNamedDeclaration) {
+        const declarations = node.declaration?.declarations;
+        if (declarations?.length !== 1) {
+          return;
+        }
+        const exportName = declarations[0].id.name;
+        if (exportName == null || !stylexProperties.has(exportName)) {
+          return;
+        }
+        stylexProperties.delete(exportName);
+      },
+
+      // Exempt used styles: export default exportStyles;
+      ExportDefaultDeclaration(node: ExportDefaultDeclaration) {
+        const exportName = node.declaration.name;
+        if (exportName == null || !stylexProperties.has(exportName)) {
+          return;
+        }
+        stylexProperties.delete(exportName);
+      },
+
+      // Exempt used styles: used as return
+      ReturnStatement(node: ReturnStatement) {
+        if (node.argument?.type === 'Identifier') {
+          const returnName = node.argument.name;
+          if (stylexProperties.has(node.argument.name)) {
+            stylexProperties.delete(returnName);
+          }
+        }
+      },
+
+      'Program:exit'() {
+        // Fallback to legacy `getSourceCode()` for compatibility with older ESLint versions
+        const sourceCode =
```

Comment:
since this is repeated across a few rules, maybe we can extract this logic out into a utils

Reply:
I think extracting getting sourcecode make sense, I will push up a PR after this PR is resolved

Reply:
[PR UP] [chore: extract eslint get souce code to util](https://github.com/facebook/stylex/pull/772)